### PR TITLE
Get entity api 

### PR DESCRIPTION
**Pull request by gearama**
to get just one entity , this api was also requested in the api review 
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

File: sdk/tables/azure-data-tables/src/tables_clients.cpp [hunk 1]
```diff
@@ -820,6 +820,37 @@ void Models::QueryEntitiesPagedResponse::OnNextPage(const Azure::Core::Context&
   *this = m_tableClient->QueryEntities(m_operationOptions, context);
 }
 
+Azure::Response<Models::TableEntity> TableClient::GetEntity(
+    const std::string& partitionKey,
+    const std::string& rowKey,
+    Core::Context const& context)
+{
+  auto url = m_url;
+  url.AppendPath(m_tableName + "(PartitionKey='" + partitionKey + "',RowKey='" + rowKey + "')");
```
Comment (antkmsft) [inline sdk/tables/azure-data-tables/src/tables_clients.cpp:829]: Don't you need to URL-encode it? URL encoding would leave most of the characters intact, but would transform, spaces, for example, to `%20`s.

**Review by antkmsft — APPROVED** [review]
LGTM, but I may not be the best expert, you may want to get Larry's review.

File: sdk/tables/azure-data-tables/inc/azure/data/tables/tables_clients.hpp [hunk 2]
```diff
@@ -301,6 +301,19 @@ namespace Azure { namespace Data { namespace Tables {
...
 
+    /**
+     * @brief Get one table entity.
+     *
+     * @param partitionKey The partition key of the entity.
+     * @param rowKey The row key of the entity.
+     * @param context for canceling long running operations.
+     * @return Entity list paged response.
+     */
+    Response<Models::TableEntity> GetEntity(
```
Comment (ahsonkhan) [inline sdk/tables/azure-data-tables/inc/azure/data/tables/tables_clients.hpp:312]: The comment doesn't match the `Response<T>`. It isn't a paged response.

File: sdk/tables/azure-data-tables/test/ut/table_client_test.cpp [hunk 3]
```diff
@@ -469,6 +469,41 @@ namespace Azure { namespace Data { namespace Test {
     EXPECT_EQ(responseQuery.TableEntities.size(), 1);
   }
 
+  TEST_P(TablesClientTest, EntityGet)
+  {
+    if (GetParam() == AuthType::Key)
+    {
+      EXPECT_TRUE(true);
+      return;
+    }
```
Comment (ahsonkhan) [inline sdk/tables/azure-data-tables/test/ut/table_client_test.cpp:478]: Why do we have this type of parameterized test if it doesn't make sense here? Deserves a comment.

File: sdk/tables/azure-data-tables/src/tables_clients.cpp [hunk 4]
```diff
@@ -820,6 +820,39 @@ void Models::QueryEntitiesPagedResponse::OnNextPage(const Azure::Core::Context&
...
+
+  Core::Http::Request request(Core::Http::HttpMethod::Get, url);
+  request.SetHeader("Accept", "application/json;odata=fullmetadata");
+
+  auto rawResponse = m_pipeline->Send(request, context);
+  auto const httpStatusCode = rawResponse->GetStatusCode();
+  if (httpStatusCode != Core::Http::HttpStatusCode::Ok)
+  {
+    throw Core::RequestFailedException(rawResponse);
+  }
```
Comment (ahsonkhan) [inline sdk/tables/azure-data-tables/src/tables_clients.cpp:841]: Do we have a test for the non-happy error path here?

**Pull request merged**